src/Symfony/Component/Messenger/Transport/Doctrine/Connection.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport\Doctrine;
+
+use Doctrine\DBAL\Connection as DriverConnection;
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\DBAL\Schema\Synchronizer\SingleDatabaseSynchronizer;
+use Doctrine\DBAL\Types\Type;
+
+/**
+ * @author Vincent Touzet <vincent.touzet@gmail.com>
+ */
+class Connection
+{
+    private $tableName;
+    private $driverConnection;
+    private $debug;
+
+    const STATUS_PENDING = 'pending';
+    const STATUS_PROCESSING = 'processing';
+    const STATUS_ACK = 'ack';
+
+    public function __construct(string $tableName, DriverConnection $driverConnection, bool $debug = false)
+    {
+        $this->tableName = $tableName;
+        $this->driverConnection = $driverConnection;
+        $this->debug = $debug;
+    }
+
+    public function publish(string $body, array $headers): void
+    {
+        if ($this->debug) {
+            $this->setup();
+        }
+        $stmt = $this->driverConnection->prepare('INSERT INTO '.$this->tableName.' (`body`, `headers`) VALUES (:body, :headers)');
+        $stmt->execute(array(
+            ':body' => $body,
+            ':headers' => \json_encode($headers),
+        ));
+    }
+
+    public function get(): ?array
+    {
+        if ($this->debug) {
+            $this->setup();
+        }
+        $stmt = $this->driverConnection->prepare("SELECT * FROM {$this->tableName} WHERE status = :status ORDER BY id ASC LIMIT 0,1");
```

Comment:
Not database agnostic. Please use your _Database Abstraction_ layer properly. Not executing hand crafted SQL. You could just use PDO if you wanted to do that.

Use query builder